Storage/SQL/SQLiteRemoteClientsAndTabs.swift

```diff
@@ -249,77 +233,51 @@ open class SQLiteRemoteClientsAndTabs: RemoteClientsAndTabs {
     }
 
     open func getClientsAndTabs() -> Deferred<Maybe<[ClientAndTabs]>> {
-        // Now find the clients.
-        let deferredClients = db.withConnection { connection -> [RemoteClient] in
-            let cursor = connection.executeQuery("SELECT * FROM \(TableClients) WHERE EXISTS (SELECT 1 FROM \(TableRemoteDevices) rd WHERE rd.guid = fxaDeviceId) ORDER BY modified DESC", factory: SQLiteRemoteClientsAndTabs.remoteClientFactory)
-            let clients = cursor.asArray()
-            cursor.close()
-            
-            return clients
-        }
-        
-        guard let clients = deferredClients.value.successValue else {
-            return deferMaybe(DatabaseError(err: deferredClients.value.failureValue as NSError?))
-        }
+        return db.withConnection { conn -> [ClientAndTabs] in
```

Comment:
Consider whether the processing work is fast enough that it's worth holding the DB connection while we do it.